pkg/api/norman/store/cluster/cluster_store.go

```diff
@@ -544,21 +551,18 @@ func (r *Store) Update(apiContext *types.APIContext, schema *types.Schema, data
 		return nil, err
 	}
 
-	// Replace rancherKubernetesEngineConfig.cloudProvider.vsphereCloudProvider.virtualCenter value to properly update virtualCenter removal
+	// When any rancherKubernetesEngineConfig.cloudProvider.vsphereCloudProvider.virtualCenter has been removed, updating cluster using k8s dynamic client
```

Comment:
I think we should add why we are doing this using the dynamic client now so a person touching this code later knows why its not using Store.Update

Reply:
Updated

Reply:
I was actually referring to why we are not using `Store.Update` but we can use Git history for that